lib/graphiti/sideload.rb

```diff
@@ -136,6 +136,12 @@ def link_filter(parents)
       base_filter(parents)
     end
 
+    def link_extra_fields
+      extra_fields = context.params.dig(:extra_fields, name)
```

Comment:
I'm not sure whether `name` is the correct attribute to use in general, it seems correct in the specs.

Reply:
This is going to be the tricky part. `name` is `Sideload#name`, or IOW the relationship name as opposed to the JSON:API type. So imagine `PositionResource` had `self.type = :special_positions`, the relationship name is `positions` but the JSON:API type is `special_positions`.

Adding a `belongs_to` spec would catch this, since the relationship name (`department`) wouldn't match the type (`departments`).

The complicating factor here is there are 2 parts - getting the fields from the url (line 140) and building the link (142). I think it makes sense that the latter is always the JSON:API type (I think here it would be `self.resource.type`). This is why the `belongs_to` spec would fail - you'd end up with `?extra_fields[department]` instead of `?extra_fields[departments]`.

But to get the fields from the URL, there are actually two few different ways you could specify this - one is the type, one is the name. The former is typical JSON:API spec, the latter is useful for fine-grained nesting (ie we want to target `positions.department.tasks.employee` not the top-level Employee or any other employee in the graph). So we'd need specs ensuring that this works however you specify it. See the specs on `query.rb` for more context if needed.

Reply:
Thanks for the pointers, I ended up added `name`, `association_name`, or `resource.type` as the potential ways to reference the related resource. Does that that make sense? Or should `association_name` and `resource.type` alone be enough?

Are there any scenarios where the `association_name` (`as:`) is set and someone would refer using the `name` directly? 🤔

Reply:
Really good question, thanks for bringing it up. I think `association_name` and `resource.type` should be enough. The use case is polymorphism, so `has_many :notes, as: :notable`. In the actual specs I believe this is `has_many :employees, as: :workspace`. I believe this will "just work" but would you mind adding a spec for this as well?

Reply:
Okay, thanks for clarifying!

There is already a test case that I think covers the `has_many :employees, as: :workspace` case:
https://github.com/graphiti-api/graphiti/blob/c866b230fe69ce08dca2eaef5116c3f32b859009/spec/serialization_spec.rb#L1345-L1356

Does that cover the same case you are referring to?

Reply:
Yep, exactly. We just want to make sure the correct link is generated for this extra fields use case. Then gtg I think!

Reply:
Ah, I understand now! The `polymorphic_belongs_to` case is what needs to be verified.
Whereby if the request has `?extra_fields[workspace]=foo` we should create a link with `?extra_fields[office]=foo` or `?extra_fields[home_office]=foo` depending on which underlying resource backs the association. 👍

Reply:
Yep I think you got it 👏 !

Reply:
Sorry for the delay 😄 

I've added the test case using the `credit_card` example as the workplace example is set up is on the `ActiveRecord` resources and not within the `PORO` ones.